source/ast/expressions/MiscExpressions.cpp

```diff
@@ -1101,6 +1101,22 @@ Expression& AssertionInstanceExpression::makeDefault(const Symbol& symbol) {
     auto result = comp.emplace<AssertionInstanceExpression>(*type, symbol, body,
                                                             /* isRecursiveProperty */ false, range);
     result->localVars = localVars.copy(comp);
+    // Check LRM 16.10 section restriction:
+    // > It's illegal to declare a local variable if it is a formal argument of a sequence
+    // declaration.
+    if (!localVars.empty() && !formalPorts.empty() && symbol.kind == SymbolKind::Sequence) {
```

Comment:
I don't think this should be restricted to just sequences. It applies to properties as well. Also I think you can place this in the createLocals() function up above in this file. There's already a map of local var names and duplicate checking there, so you can avoid the O(n^2) algorithm here.

Reply:
fixed